eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml

```diff
@@ -82,6 +84,11 @@ stages:
               testResultsFormat: 'VSTest'
               mergeTestResults: true
 
+      - job: CustomTests
```

Comment:
Does this need to be a separate job?

For our purposes here I suspect just adding a new parameter that allows for a ci.yml file to add additional steps to this build job would probably be easier.